instrumentation/src/javatests/org/oppia/android/instrumentation/topic/TopicActivityTest.kt

```diff
@@ -0,0 +1,71 @@
+package org.oppia.android.instrumentation.topic
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByText
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.scrollRecyclerViewTextIntoView
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.startOppiaFromScratch
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForText
+
+/** Tests for TopicActivity. */
+class TopicActivityTest {
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testTopicActivity_exploredFully_finishedSuccessfully() {
+    navigateToTopicActivity()
+    device.findObjectByText("See More").click()
+    device.waitForText("See Less")
+    device.findObjectByText("See Less").click()
+
+    device.findObjectByText("LESSONS").click()
+    device.findObjectByRes("expand_list_icon").click()
+    device.waitForRes("chapter_list_container")
+    device.findObjectByText("What is a Fraction?").click()
+    device.waitForText("CONTINUE")
```

Comment:
Continue button might get beyond the screen size. check if exploration specific resource id is visible or not

Reply:
Can you explain this a bit more? If this means to use `waitForRes` instead then done.